yarn-project/acir-simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -0,0 +1,83 @@
+import { Fr } from '@aztec/foundation/fields';
+
+import { AvmContext } from '../avm_context.js';
+import { AvmMachineState } from '../avm_machine_state.js';
+import { AvmJournal } from '../journal/journal.js';
+import { Instruction } from './instruction.js';
+
+export class Call extends Instruction {
+  static type: string = 'CALL';
+  static numberOfOperands = 7;
+
+  constructor(
+    private /* Unused due to no formal gas implementation at this moment */ _gasOffset: number,
+    private addrOffset: number,
+    private argsOffset: number,
+    private argSize: number,
+    private retOffset: number,
+    private retSize: number,
+    private successOffset: number,
+  ) {
+    super();
+  }
+
+  // TODO(https://github.com/AztecProtocol/aztec-packages/issues/3992): there is no concept of remaining / available gas at this moment
+  async execute(machineState: AvmMachineState, journal: AvmJournal): Promise<void> {
```

Comment:
IIUC this does not conform to the `execute` in the `Instruction` interface. In particular the interpreter `run()` is not awaiting on `execute`.

Reply:
Ah yes correct, I will need to make everything async, will update